src/push.c

```diff
@@ -349,8 +349,8 @@ static int queue_objects(git_push *push)
 		if (git_oid_is_zero(&head->oid))
 			continue;
 
-		/* TODO */
-		git_revwalk_hide(rw, &head->oid);
+		if ((error = git_revwalk_hide(rw, &head->oid)) < 0 && error != GIT_ENOTFOUND)
```

Comment:
I think that we need to filter out other non-errors here.  (object is not a commit)  But otherwise, I 👍 this PR once CI is green.

Reply:
Easier said than done. Quoting `git_revwalk__push_commit`, which is invoked by `git_revwalk_hide`:

```
error = git_object_peel(&obj, oobj, GIT_OBJECT_COMMIT);
	git_object_free(oobj);

	if (error == GIT_ENOTFOUND || error == GIT_EINVALIDSPEC || error == GIT_EPEEL) {
		/* If this comes from e.g. push_glob("tags"), ignore this */
		if (opts->from_glob)
			return 0;

		git_error_set(GIT_ERROR_INVALID, "object is not a committish");
		return -1;
	}
```

Reply:
I'll just update the return statement to return `error` verbosely.